### PR TITLE
feat(stripe): Upgrade to SDK v4

### PR DESCRIPTION
**Pull request by disq**
Closes https://github.com/cloudquery/cloudquery/issues/11933

### CODE REVIEW

File: plugins/source/stripe/client/testing.go [hunk 1]
```diff
@@ -45,32 +43,33 @@ func MockTestHelper(t *testing.T, table *schema.Table, opts TestOptions) {
 		}
 	}()
 
-	newTestExecutionClient := func(ctx context.Context, logger zerolog.Logger, spec specs.Source, opts source.Options) (schema.ClientMeta, error) {
-		var stSpec Spec
-		if err := spec.UnmarshalSpec(&stSpec); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal stripe spec: %w", err)
-		}
+	sched := scheduler.NewScheduler(scheduler.WithLogger(logger))
+	spec := &Spec{}
```
Comment (candiduslynx) [inline plugins/source/stripe/client/testing.go:47]: don't you need to set up `APIKey` as you should err out on the `Spec.Validate`?

File: plugins/source/stripe/client/testing.go [hunk 2]
```diff
@@ -45,32 +43,33 @@ func MockTestHelper(t *testing.T, table *schema.Table, opts TestOptions) {
...
-	p.SetLogger(logger)
-	source.TestPluginSync(t, p, specs.Source{
-		Name:         "dev",
-		Path:         "cloudquery/dev",
-		Version:      version,
-		Tables:       []string{table.Name},
-		Destinations: []string{"mock-destination"},
-	})
+func filterInserts(msgs message.SyncMessages) message.SyncInserts {
+	inserts := []*message.SyncInsert{}
```
Comment (candiduslynx) [inline plugins/source/stripe/client/testing.go:66]: I feel like this should be a part of SDK as I saw this in several PRs already

File: plugins/source/stripe/go.mod [hunk 3]
```diff
@@ -4,18 +4,20 @@ go 1.19
 
 require (
 	github.com/apache/arrow/go/v13 v13.0.0-20230630125530-5a06b2ec2a8e
-	github.com/cloudquery/plugin-pb-go v1.6.0
-	github.com/cloudquery/plugin-sdk/v3 v3.10.6
+	github.com/cloudquery/plugin-sdk v1.45.0
```
Comment (candiduslynx) [inline plugins/source/stripe/go.mod:7]:
```suggestion
```

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/source/stripe/client/testing.go (hunk 2 above)
Comment (disq) [reply plugins/source/stripe/client/testing.go:66]: it is, but it was renamed

File: plugins/source/stripe/client/testing.go (hunk 1 above)
Comment (disq) [reply plugins/source/stripe/client/testing.go]: fixed

File: plugins/source/stripe/go.sum [hunk 4]
```diff
@@ -43,10 +43,12 @@ github.com/cloudquery/arrow/go/v13 v13.0.0-20230703001435-df3b664a289d h1:Rm71jc
 github.com/cloudquery/arrow/go/v13 v13.0.0-20230703001435-df3b664a289d/go.mod h1:W69eByFNO0ZR30q1/7Sr9d83zcVZmF2MiP3fFYAWJOc=
 github.com/cloudquery/plugin-pb-go v1.6.0 h1:4g+tPhRAQUWpGyQBMcj6D3yVwKqwEN3nynv+mlBysQo=
 github.com/cloudquery/plugin-pb-go v1.6.0/go.mod h1:R0Wse6NbJDZIHcRQjJ1sZGYDo3mrIDm4k3El1YUrvGA=
+github.com/cloudquery/plugin-sdk v1.45.0 h1:5vrfQZtaO1dp6ebKt8ouXDmPC7eeLuOB3JMd+FTRSYk=
```
Comment (candiduslynx) [inline plugins/source/stripe/go.sum:46]: this requires `go mod tidy`

**Pull request merged**